inc/vcf/record_cache.hpp

```diff
@@ -89,17 +94,39 @@ namespace ebi
                     duplicates.emplace_back(new DuplicationError{record_core.line, ss.str(), duplicate_variant_lines});
                 }
 
-                cache.insert(range.second, record_core);
+                // Insert the record in cache if the allele isn't a symbolic variant
+                if( is_symbolic_allele(record_core) == symbolic_allele_check) {
```

Comment:
we found this a bit confusing. With the current design, each record is checked against the regular cache and the symbolic cache, but only inserted in the corresponding cache. My suggestion is 

1) in the ragel part we aim for this: 
```
cache.check_duplicates(*record);

for r in cache.get_duplicates
    handle_error(r)

for r in cache.get_symbolic_duplicates
    handle_warning(r)
```

2) adding two fields `duplicates` and `symbolic_duplicates`, 
3) the function `check_duplicates` checks `.equal_range(record_core);` against one cache or the other depending on whether the recordCore is symbolic or not.
4) using the same distiction, emplace duplicates in `duplicates` or `symbolic_duplicates`.
4) the function `check_duplicates` doesn't return anything.
5) add two getters: `get_duplicates()` and `get_symbolic_duplicates()`